mbuild/box.py

```diff
@@ -2,151 +2,357 @@
-        Maximum x, y, z coordinates.
-    lengths : np.ndarray, shape(3,), dtype=float
-        Box length in x, y and z directions.
-    angles : np.ndarray, shape(3,), dtype=float, default=[90,90,90]
-        Angles defining the tilt of the box
+    box_vectors : np.ndarray, shape=(3,3), dtype=float
+        Vectors that define the parallelepiped (Box).
+    Lx : float
+        Length of the Box in the x dimension
+    Ly : float
+        Length of the Box in the y dimension
+    Lz : float
+        Length of the Box in the z dimension
+    xy : float
+        Tilt factor needed to displace an orthogonal box's xy face to its parallelepiped structure.
+    xz : float
+        Tilt factor needed to displace an orthogonal box's xz face to its parallelepiped structure.
+    yz : float
+        Tilt factor needed to displace an orthogonal box's yz face to its parallelepiped structure.
+    precision : int
+        Precision of the floating point numbers when accessing the __repr__
 
+    NOTE
+    ----
+    Box vectors are expected to be provided in row-major format.
     """
-    def __init__(self, lengths=None, mins=None, maxs=None, angles=None):
-        if lengths is None:
-            if mins is None or maxs is None:
-                raise ValueError(
-                    "Either provide `lengths` or `mins` and `maxs`. "
-                    "You provided: "
-                    "lengths={} mins={} maxs={}".format(lengths, mins, maxs)
-                )
-            mins = np.array(mins, dtype=float)
-            maxs = np.array(maxs, dtype=float)
-            assert mins.shape == (3, ), "Given mins have wrong dimensions"
-            assert maxs.shape == (3, ), "Given maxs have wrong dimensions"
-            assert all(mins <= maxs), "Given mins are greater than maxs"
-            self._mins = _BoxArray(array=mins, var="mins", box=self)
-            self._maxs = _BoxArray(array=maxs, var="maxs", box=self)
-            self._lengths = _BoxArray(array=(self.maxs - self.mins), var="lengths", box=self)
-        else:
-            if mins is not None or maxs is not None:
-                warn(
-                    "Provided `lengths` and `mins` and/or `maxs`. Only `lengths` "
-                    "is being used. You provided: "
-                    "lengths={} mins={} maxs={}".format(lengths, mins, maxs)
-                )
-            if isinstance(lengths, int) or isinstance(lengths, float):
-                lengths = np.array(lengths*np.ones(3), dtype=float)
-            else:
-                lengths = np.array(lengths, dtype=float)
-            assert lengths.shape == (3, )
-            assert all(lengths >= 0), "Given lengths are negative"
-            self._mins = _BoxArray(array=(0,0,0), var="mins", box=self)
-            self._maxs = _BoxArray(array=lengths, var="maxs", box=self)
-            self._lengths = _BoxArray(array=lengths, var="lengths", box=self)
+
+    def __init__(self, lengths=None, angles=None, precision=None):
```

Comment:
```suggestion
    def __init__(self, lengths, angles=None, precision=None):
```
let's not provide a default for lengths--the error message is not as helpful
```
box = mb.Box()

>>> TypeError: cannot unpack non-iterable NoneType object
```
compared to 
```
>>> TypeError: Box() missing 1 required positional argument: 'lengths'
```
also, as a required argument it will fail faster :)